src/components/ClusterNodes/ClusterList.tsx

```diff
@@ -0,0 +1,204 @@
+            clearSearch()
+        }
+    }
+
+    const renderSearch = (): JSX.Element => {
+        return (
+            <div className="search position-rel margin-right-0 en-2 bw-1 br-4 h-32">
+                <Search className="search__icon icon-dim-18" />
+                <input
+                    type="text"
+                    placeholder="Search clusters"
+                    value={searchText}
+                    className="search__input"
+                    onChange={(event) => {
+                        setSearchText(event.target.value)
+                    }}
+                    onKeyDown={handleFilterKeyPress}
+                />
+                {searchApplied && (
+                    <button className="search__clear-button" type="button" onClick={clearSearch}>
+                        <Clear className="icon-dim-18 icon-n4 vertical-align-middle" />
+                    </button>
+                )}
+            </div>
+        )
+    }
+
+    if (loader) {
+        return <Progressing pageLoader />
+    }
+
+    return (
+        <>
+            <PageHeader headerName="Clusters" />
+            <div className={`cluster-list bcn-0 ${noResults ? 'no-result-container' : ''}`}>
+                <div className="flexbox content-space pl-20 pr-20 pt-16 pb-16">
+                    {renderSearch()}
+                    <div className="fs-13">
+                        {lastDataSyncTimeString && (
+                            <span>
+                                {lastDataSyncTimeString}{' '}
+                                <button className="btn btn-link p-0 fw-6 cb-5 ml-5 fs-13" onClick={getData}>
+                                    Refresh
+                                </button>
+                            </span>
+                        )}
+                    </div>
+                </div>
+                {noResults ? (
+                    <ClusterNodeEmptyState actionHandler={clearSearch} />
+                ) : (
+                    <div className="" style={{ minHeight: 'calc(100vh - 125px)' }}>
+                        <div className="cluster-list-row fw-6 cn-7 fs-12 border-bottom pt-8 pb-8 pr-20 pl-20 text-uppercase">
+                            <div>Cluster</div>
+                            <div>Connection status</div>
+                            <div>Nodes</div>
+                            <div>NODE Errors</div>
+                            <div>K8S version</div>
+                            <div>CPU Capacity</div>
+                            <div>Memory Capacity</div>
+                        </div>
+                        {filteredClusterList?.map((clusterData) => {
+                            const errorCount = clusterData.nodeErrors ? Object.keys(clusterData.nodeErrors).length : 0
+                            return (
+                                <div className="cluster-list-row fw-4 cn-9 fs-13 border-bottom-n1 pt-12 pb-12 pr-20 pl-20 hover-class">
```

Comment:
Suggestion - Extract this to a separate function & then use that function here.

Reply:
Done